soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -191,12 +202,35 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	idc = device_get_binding(DT_LABEL(DT_INST(0, intel_cavs_idc)));
 #endif
 
+	/* Clear any previous IDC message interrupts */
+	idc_reg = idc_read(IPC_IDCIETC(cpu_num), 0);
+	if (idc_reg & IPC_IDCIETC_DONE)
+		idc_write(IPC_IDCIETC(cpu_num), 0, idc_reg);
```

Comment:
Not sure all this is needed, it looks like an attempt to chase races with ROM initialization?   In theory, we shouldn't ever enter this code unless with know with 100% certainty that the target CPU has power and is spinning in the ROM initialization loop waiting on an IDC interrupt.  If we don't know that, there's really no way to do this reliably anyway.

My guess is that this all goes away with a proper host/DSP power management protocol.

Reply:
@andyross Some of that indeed isn't needed, I removed it. But some changes I haven't been able to get rid of. Need to try a bit harder or see if this also works for other Xtensa SMP tests